server/src/main/java/org/elasticsearch/common/geo/Extent.java

```diff
@@ -165,15 +167,15 @@ void writeCompressed(ByteBuffersDataOutput output) throws IOException {
         switch (type) {
             case NONE_SET : break;
             case POSITIVE_SET:
-                output.writeVInt(this.posRight);
+                output.writeVInt(this.posLeft);
                 output.writeVLong((long) this.posRight - this.posLeft);
                 break;
             case NEGATIVE_SET:
-                output.writeInt(this.negRight);
+                output.writeVInt(-this.negRight);
                 output.writeVLong((long) this.negRight - this.negLeft);
                 break;
             case CROSSES_LAT_AXIS:
-                output.writeInt(this.posRight);
+                output.writeVInt(this.posRight);
                 output.writeVLong((long) this.posRight - this.negLeft);
```

Comment:
should we encode negLeft directly instead of its delta with posRight, i.e. `output.writeVInt(-this.negLeft);`

Reply:
I suppose the range of values are comparable for both. I'll update to write VInt. thanks for the suggestion!